nni/algorithms/compression/v2/pytorch/pruning/basic_scheduler.py

```diff
@@ -75,12 +83,71 @@ def pruning_one_step(self, task: Task) -> TaskResult:
                 self.pruner._unwrap_model()
 
         # evaluate
-        score = self.evaluator(compact_model) if self.evaluator is not None else None
+        if self.evaluator is not None:
+            if self.speed_up:
+                score = self.evaluator(compact_model)
+            else:
+                self.pruner._wrap_model()
+                score = self.evaluator(compact_model)
+                self.pruner._unwrap_model()
+        else:
+            score = None
 
         # clear model references
         self.pruner.clear_model_references()
 
         return TaskResult(task.task_id, compact_model, compact_model_masks, pruner_generated_masks, score)
 
+    def pruning_one_step_reset_weight(self, task: Task) -> TaskResult:
+        """
+        finetune -> generate masks -> reset weight -> speed up -> evaluate
```

Comment:
Why the `normal` way and the `reset` way have two different workflow?

Reply:
because in `lottery ticket` paper, the workflow is finetuning at first, then pruning and reset the weight to the origin, the final output can be seen as a structure and don't care about the weight values.

In other pruning paper, usually pruning at first then finetuning the pruned model to recover the acc.